src/Modules/TasksModule/components/TaskData/TaskData.tsx

```diff
@@ -1,5 +1,301 @@
+  const navigate = useNavigate();
+  const location = useLocation();
+  const task = location.state;
+  // console.log(location);
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<TaskSubmitInterface>({
+    criteriaMode: "all",
+  });
+
+  const onSubmit: SubmitHandler<TaskSubmitInterface> = async (task) => {
+    try {
+      await axios.post(`${baseUrl}/Task`, task, {
+        headers: getRequestHeaders(),
+      });
+      toast.success("New Task Has Been Added Successfully");
+      navigate("/dashboard/tasks");
+    } catch (error) {
+      handleApiError(error);
+    }
+  };
+
+  const getProject = async (pageSize: number) => {
+    setIsLoading(true);
+    try {
+      const { data } = await axios.get(
+        `${baseUrl}/Project?pageSize=${pageSize}&pageNumbe=${pageNumber}`,
+        {
+          headers: getRequestHeaders(),
+          // params: {
+          //   name: name,
+          // },
+        }
+      );
+      setprojects(data.data);
+    } catch (err) {
+      handleApiError(err);
+    }
+    setIsLoading(false);
+  };
+  const onUpdateSubmit: SubmitHandler<TaskUpdateSubmitInterface> = async (
+    data
+  ) => {
+    try {
+      await axios.put(`${baseUrl}/Task/${taskId}`, data, {
+        headers: getRequestHeaders(),
+      });
+      // handleCloseUpdate();
+      // getTask(taskName, 10, 1);
+      toast.success(`Updated Task Successfully`);
+      navigate("/dashboard/tasks");
+    } catch (error) {
+      handleApiError(error);
+    }
+    // console.log(data);
+  };
+  const getUsers = async (pageSize: number) => {
+    setIsLoading(true);
+    try {
+      const { data } = await axios.get(
+        `${baseUrl}/Users?pageSize=${pageSize}&pageNumbe=${pageNumber}`,
+        {
+          headers: getRequestHeaders(),
+          // params: {
+          //   name: name,
+          // },
+        }
+      );
+      setUsers(data.data);
+      console.log(data.data);
+    } catch (err) {
+      handleApiError(err);
+    }
+    setIsLoading(false);
+  };
+
+  const btnloading = () => {
+    return (
+      <div className="loader">
+        <i>&lt;</i>
+        <span>LOADING</span>
+        <i>/&gt;</i>
+      </div>
+    );
+  };
+
+  useEffect(() => {
+    if (task) {
+      setIsUpdate(true);
+      setTaskId(task.id);
+      reset({
+        title: task.title,
+        description: task.description,
+        employeeId: task.employee.id,
+      });
+    }
+    getProject(10);
+    getUsers(10);
+  }, []);
+
+  return (
+    <>
+      <section>
+        <div
+          className={`project-data-head container-fluid shadow-sm  head-bg pt-5 pb-4 px-5`}
+        >
+          <div className={`row`}>
+            <div className="col-md-6">
+              <div
+                className={`text-theme text-lg-start text-sm-center text-center`}
+              >
+                <button
+                  onClick={() => {
+                    navigate("/dashboard/tasks");
```

Comment:
What is the difference between having a Link and using a button for only navigation?